src/pages/ScreenRouting/mediaScreens/MediaEvent/MediaEvent.tsx

```diff
@@ -0,0 +1,192 @@
+import usePlaylist from '#src/hooks/usePlaylist';
+import useEntitlement from '#src/hooks/useEntitlement';
+import StartWatchingButton from '#src/containers/StartWatchingButton/StartWatchingButton';
+import Cinema from '#src/containers/Cinema/Cinema';
+import useBreakpoint, { Breakpoint } from '#src/hooks/useBreakpoint';
+import TrailerModal from '#src/containers/TrailerModal/TrailerModal';
+import ShareButton from '#components/ShareButton/ShareButton';
+import FavoriteButton from '#src/containers/FavoriteButton/FavoriteButton';
+import PlayTrailer from '#src/icons/PlayTrailer';
+import Button from '#components/Button/Button';
+import useQueryParam from '#src/hooks/useQueryParam';
+import InlinePlayer from '#src/containers/InlinePlayer/InlinePlayer';
+import StatusIcon from '#components/StatusIcon/StatusIcon';
+
+const MediaEvent: ScreenComponent<PlaylistItem> = ({ data, isLoading }) => {
+  const { t, i18n } = useTranslation('video');
+
+  const [playTrailer, setPlayTrailer] = useState<boolean>(false);
+  const breakpoint = useBreakpoint();
+
+  // Routing
+  const navigate = useNavigate();
+
+  const params = useParams();
+  const id = params.id || '';
+  const play = useQueryParam('play') === '1';
+  const feedId = useQueryParam('r');
+
+  // Config
+  const { config, accessModel } = useConfigStore(({ config, accessModel }) => ({ config, accessModel }), shallow);
+  const { siteName, features, custom } = config;
+
+  const isFavoritesEnabled: boolean = Boolean(features?.favoritesList);
+  const inlineLayout = Boolean(custom?.inlinePlayer);
+
+  // Media
+  const { isLoading: isTrailerLoading, data: trailerItem } = useMedia(data?.trailerId || '');
+  const { isLoading: isPlaylistLoading, data: playlist } = usePlaylist(feedId || '');
+
+  // Event
+  const liveEvent = useLiveEvent(data);
+
+  // User, entitlement
+  const { user, subscription } = useAccountStore(({ user, subscription }) => ({ user, subscription }), shallow);
+  const { isEntitled } = useEntitlement(data);
+
+  // Handlers
+  const goBack = () => data && navigate(mediaURL(data, feedId, false));
+  const onCardClick = (item: PlaylistItem) => navigate(mediaURL(item, feedId));
+
+  const handleComplete = useCallback(() => {
+    if (!id || !playlist) return;
+
+    const index = playlist.playlist.findIndex(({ mediaid }) => mediaid === id);
+    const nextItem = playlist.playlist[index + 1];
+
+    if (nextItem.mediaStatus === MediaStatus.SCHEDULED) {
+      return;
+    }
+
+    return nextItem && navigate(mediaURL(nextItem, feedId, true));
+  }, [id, playlist, navigate, feedId]);
+
+  // Effects
+  useEffect(() => {
+    (document.scrollingElement || document.body).scroll({ top: 0 });
+  }, [id]);
+
+  // UI
+  const pageTitle = `${data.title} - ${siteName}`;
+  const canonicalUrl = data ? `${window.location.origin}${mediaURL(data)}` : window.location.href;
+
+  const primaryMetadata = (
+    <>
+      <StatusIcon mediaStatus={data.mediaStatus} />
```

Comment:
This now uses the `data.mediaStatus` and only updates when the media item gets refreshed and transformed. I don't think this will ever switch when the state changes when already on this page, right?

Reply:
It will update. The usePlaylist will update the media item.